internal/controller/pkg/controller/runtime.go

```diff
@@ -1,5 +1,50 @@
 package controller
 
+// ActiveRuntime is the runtime to use for packages with runtime.
+type ActiveRuntime struct {
+	runtimes       map[string]PackageRuntime
+	defaultRuntime PackageRuntime
+}
+
+// RuntimeOption defines options to build up an active runtime.
+type RuntimeOption func(runtime *ActiveRuntime)
+
+// WithDefaultPackageRuntime marks a default runtime for unset kinds.
+func WithDefaultPackageRuntime(runtime PackageRuntime) RuntimeOption {
+	return func(ar *ActiveRuntime) {
+		ar.defaultRuntime = runtime
+	}
+}
+
+// WithPackageRuntime associates a runtime to a kind.
+func WithPackageRuntime(kind string, runtime PackageRuntime) RuntimeOption {
+	return func(ar *ActiveRuntime) {
+		ar.runtimes[kind] = runtime
+	}
+}
+
+// NewActiveRuntime builds an ActiveRuntime based on the provided options.
+func NewActiveRuntime(o ...RuntimeOption) ActiveRuntime {
+	r := ActiveRuntime{
+		runtimes: make(map[string]PackageRuntime),
+	}
+	for _, o := range o {
+		o(&r)
+	}
+	if r.defaultRuntime == PackageRuntimeUnspecified {
+		r.defaultRuntime = PackageRuntimeDeployment
+	}
+	return r
+}
```

Comment:
It could be simplified a bit as follows I believe:

```suggestion
func NewActiveRuntime(o ...RuntimeOption) ActiveRuntime {
	r := ActiveRuntime{
		runtimes: make(map[string]PackageRuntime),
		defaultRuntime: PackageRuntimeDeployment
	}
	for _, o := range o {
		o(&r)
	}

	return r
}
```